packages/dashboards/addon/routes/dashboards/dashboard/widgets/widget.js

```diff
@@ -7,6 +7,27 @@ import { getOwner } from '@ember/application';
 import ReportRoute from 'navi-reports/routes/reports/report';
 
 export default ReportRoute.extend({
+  /**
+   * Decide if we need to store the query params of the parent route so we can add them to breadcrumbs
+   * @param {Ember.Transition} transition - current route transition.
+   */
+  beforeModel(transition) {
+    const controller = this.controllerFor(this.routeName);
+    const fromRoute = transition.from && transition.from.find(info => info.paramNames.includes('dashboard_id'));
+    const fromDashboardId = fromRoute ? fromRoute.params.dashboard_id : null;
+    const toRoute = transition.to && transition.to.find(info => info.paramNames.includes('dashboard_id'));
+    const toDashboardId = toRoute ? toRoute.params.dashboard_id : null;
+
+    if (
+      transition.from &&
+      transition.from.name.startsWith('dashboards.dashboard') &&
+      fromDashboardId === toDashboardId
+    ) {
+      controller.set('parentQueryParams', transition.from.queryParams);
```

Comment:
Instead of handing query params in sub routes, could we cache them in the route that has the query params?

Reply:
Unfortunately no, that lives in the view route which is not a child of the widget route. The information doesn't need to be cached as it's already given in the transition.  Also the management of cache here would be troublesome. The only reason it's put in the controller is so that the the template has access to it.

Reply:
I was thinking the conditional clearing the query params would be similar to the way you did conditional rollback of the model. This way any child/sibling route would not have to be knowledgeable about the `dashboard.view` route params and have all the `link-to`s correct